identity/client/src/pages/login/LoginPage.tsx

```diff
@@ -41,18 +69,31 @@ const LoginForm: React.FC<LoginFormProps> = ({ onSuccess }) => {
         name="username"
         value={username}
         onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-          setUsername(e.target.value)
+          setUsername(e.target.value.trim())
         }
         labelText={t("username")}
-        invalid={false}
-        invalidText={undefined}
+        invalid={usernameError.hasError}
+        invalidText={usernameError.errorText}
         placeholder={t("username")}
+        onBlur={({ target }) => {
+          if (target.value.trim().length < 1) {
+            setUsernameError({
+              hasError: true,
+              errorText: "Username field is required",
+            });
+          } else {
+            setUsernameError({
+              hasError: false,
+              errorText: "",
+            });
+          }
+        }}
```

Comment:
We don't have this behavior on Tasklist or Operate. I think you should check it if it's not better to be consistent across apps.

IMO we should just remove it

Reply:
This comment is wrong, we do validate on blur in Operate and Tasklist